git/internal/e2e/github_test.go

```diff
@@ -61,12 +71,45 @@ func TestGitHubE2E(t *testing.T) {
 	if githubOrgname == "" {
 		t.Fatalf("could not read github org name")
 	}
+	appID := os.Getenv(githubAppIDEnv)
+	if appID == "" {
+		t.Fatalf("could not read github app id")
+	}
+	githubAppID, err = strconv.Atoi(appID)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	installID := os.Getenv(githubAppInstallIDEnv)
+	if installID == "" {
+		t.Fatalf("could not read github app installation id")
+	}
+	githubAppInstallationID, err = strconv.Atoi(installID)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	githubAppPrivateKey := []byte(os.Getenv(githubAppPKEnv))
+	if len(githubAppPrivateKey) == 0 {
+		t.Fatalf("could not read github app private key")
+	}
 
 	c, err := github.NewClient(gitprovider.WithDestructiveAPICalls(true), gitprovider.WithOAuth2Token(githubOAuth2Token))
 	g.Expect(err).ToNot(HaveOccurred())
 	orgClient := c.OrgRepositories()
 
-	repoInfo := func(proto git.TransportType, repo gitprovider.OrgRepository) (*url.URL, *git.AuthOptions, error) {
+	givePermissionsToApp := func(repo gitprovider.OrgRepository) error {
+		ctx := context.Background()
+		githubClient := c.Raw().(*gogithub.Client)
+		ghRepo, _, err := githubClient.Repositories.Get(ctx, githubOrgname, repo.Repository().GetRepository())
```

Comment:
@stefanprodan, would appreciate your feedback on the below - 

To test github app authentication, the app installation needs to be given access to the new repository that the test creates. The API used to do this is in the [go-github](github.com/google/go-github/v64/github) package, specifically [AddRepository](https://pkg.go.dev/github.com/google/go-github/v64@v64.0.0/github#AppsService.AddRepository). While reviewing the code I noticed that all go-github APIs are abstracted in fluxcd's [git-provider](https://github.com/fluxcd/go-git-providers/gitprovider) and we do not take a direct dependency on go-github in pkg. 

Question: 
- This test code uses the Raw interface exposed by go-git-providers and will add a direct dependency on go-github. Before I delve deep into go-git-providers code, I wanted to check if this is ok for test purposes or is it recommended to expand go-git-providers to expose the new API ?
- Today go-git-providers does not have a notion of Apps. I am yet to fully understand the code structure for go-git-providers, but any thoughts on the right place we can expose the API from ? 

Thanks for your inputs!

Reply:
For this PR I think it's Ok to use go-github directly. After GitHub Apps are implemented in source-controller, we need to extend `flux bootstrap github` to support Apps and then we'll need to modify go-git-providers.

Reply:
> To test github app authentication, the app installation needs to be given access to the new repository that the test creates.

The GitHub App that I created should have access to all repos, so I'm not sure why we need to give it access for new repos.

Reply:
When a github app is installed, the permissions can be given to all repos or specific repos. If we want to run the end to end tests with restricted permissions (for security reasons), then this logic is needed. For example, if someone is running end-to-end tests in their environment and does not want to give full write access to all repos.